Promote x64-windows-release out of community

BillyONeal opened the pull request:
as requested/sponsored by Azure Core / Storage @vaboca

Azure Storage is sponsoring the 3 machines of Azure Quota to turn this on.

BillyONeal committed:
Promote x64-windows-release out of community as requested/sponsored by Azure Core / Storage.

Drive-by: enable dbghelp

BillyONeal committed:
Typos

BillyONeal committed:
Skip the dbghelps that don't work.

Neumann-A commented:
You might also need to deactivate a lot of tests from scripts/tests since they are not designed to run in single config mode.

BillyONeal commented:
> You might also need to deactivate a lot of tests from scripts/tests since they are not designed to run in single config mode.

Probably, but I'm going to let the hamsters tell me which ones :)

ras0219-msft commented on scripts/ci.baseline.txt:
```diff
@@ -130,6 +136,7 @@ lapack-reference:x64-osx=skip
 lapack-reference:x64-uwp=skip
 #openblas:arm64-osx=skip
 #openblas:x64-osx=skip
+#openblas:x64-windows-release=skip
```
I don't see an `#openblas:x64-windows=skip`

ras0219-msft commented on scripts/ci.baseline.txt:
```diff
@@ -1296,104 +1344,116 @@ vcpkg-ci-gdal:arm64-android=pass
 vcpkg-ci-paraview:x64-linux=pass
 vcpkg-ci-paraview:x64-osx=pass
-vcpkg-ci-paraview:arm64-osx=pass
+vcpkg-ci-paraview:x64-windows-release=pass
```
Was the removal of `vcpkg-ci-paraview:arm64-osx=pass` intentional?

BillyONeal commented:
```
REGRESSION: cmake-user:x64-windows-release failed with BUILD_FAILED. If expected, add cmake-user:x64-windows-release=fail to D:\a\_work\1\s\scripts\azure-pipelines/../ci.baseline.txt.
REGRESSION: dbghelp:x64-windows-release failed with BUILD_FAILED. If expected, add dbghelp:x64-windows-release=fail to D:\a\_work\1\s\scripts\azure-pipelines/../ci.baseline.txt.
REGRESSION: openvino:x64-windows-release failed with BUILD_FAILED. If expected, add openvino:x64-windows-release=fail to D:\a\_work\1\s\scripts\azure-pipelines/../ci.baseline.txt.
REGRESSION: qtwebengine:x64-windows-release failed with BUILD_FAILED. If expected, add qtwebengine:x64-windows-release=fail to D:\a\_work\1\s\scripts\azure-pipelines/../ci.baseline.txt.
REGRESSION: qt:x64-windows-release cascaded, but it is required to pass. (D:\a\_work\1\s\scripts\azure-pipelines/../ci.baseline.txt).
REGRESSION: unit-test-cmake:x64-windows-release failed with BUILD_FAILED. If expected, add unit-test-cmake:x64-windows-release=fail to D:\a\_work\1\s\scripts\azure-pipelines/../ci.baseline.txt.
```

BillyONeal committed:
Remove wrong openblas line.

BillyONeal committed:
Revert dbghelp changes entirely.

BillyONeal committed:
Skip x64-windows-release dbghelp and sort.

BillyONeal committed:
Actually fix the triplet switch handling.

BillyONeal committed:
Mark qtwebengine as requiring 'native'

There is a bug in this port where non-native builds propagate the standard language mode incorrectly

BillyONeal committed:
Merge remote-tracking branch 'origin/master' into HEAD

# Conflicts:
#	scripts/ci.baseline.txt

BillyONeal committed:
[cmake-user] Support single-config.

BillyONeal committed:
Fix unit-test-cmake to support release-only.

BillyONeal commented:
> > You might also need to deactivate a lot of tests from scripts/tests since they are not designed to run in single config mode.
> 
> Probably, but I'm going to let the hamsters tell me which ones :)

It looks like most of them worked OK

BillyONeal committed:
Curse sadly significant whitespace.

BillyONeal committed:
Merge remote-tracking branch 'origin/master' into promote-release-only

# Conflicts:
#	ports/qtwebengine/vcpkg.json
#	versions/baseline.json
#	versions/q-/qtwebengine.json

BillyONeal commented:
Only build failure is openvino which is a baseline issue.

BillyONeal committed:
Merge remote-tracking branch 'origin/master' into promote-release-only

# Conflicts:
#	scripts/ci.baseline.txt

cenit commented on scripts/ci.baseline.txt:
```diff
@@ -1291,105 +1342,116 @@ vcpkg-ci-gdal:arm64-android=pass
-vcpkg-ci-openimageio:x64-osx=pass
+vcpkg-ci-openimageio:x86-windows=pass
+vcpkg-ci-openimageio:x64-windows=pass
+vcpkg-ci-openimageio:x64-windows-static=pass
```
x64-windows-static for openimageio is double, there is no -release one. Typo?

BillyONeal replied:
Good catch! Will drive-by fix in https://github.com/microsoft/vcpkg/pull/43858

cenit replied:
i always read your diffs.
Always
👀 

🤣
